app/api.go

```diff
@@ -45,6 +46,24 @@ func (lm *LearningMaterialAPI) request(next http.Handler) http.HandlerFunc {
 			return
 		}
 
+		if lm.conf.API.Captcha.Enabled {
+			chalsEncoded := b64.StdEncoding.EncodeToString([]byte(r.URL.Query().Get(requestedChallenges)))
+			_, err = r.Cookie(chalsEncoded)
+			if err != nil {
+				isValid := lm.captcha.Verify(r.FormValue("g-recaptcha-response"))
+				if !isValid {
+					formActionURL := fmt.Sprintf("/api/?%s=%s", requestedChallenges, r.URL.Query().Get(requestedChallenges))
+
+					w.WriteHeader(http.StatusBadRequest) //todo make 404 not found page
+					w.Header().Set("Content-Type", "text/html; charset=utf-8")
+					_, _ = w.Write([]byte(getCaptchaPage(formActionURL, lm.conf.API.Captcha.SiteKey)))
+					return
+				}
+				authC := http.Cookie{Name: chalsEncoded, Value: r.FormValue("g-recaptcha-response"), Path: "/", MaxAge: 200}
+				http.SetCookie(w, &authC)
+			}
+		}
+
```

Comment:
It's actually a bit weird. but that was the only possible option to implement it

Reply:
the code first checks if the captcha is enabled, 
then it check for a cookie based on the requested challenges,
if the cookie exists just goes to the next request (waiting page),
if not check if the captcha is correct, if yes goes to the next request (waiting page),
if not just shows the recaptcha page

Reply:
the cookie expires after 200 seconds